src/chunk.c

```diff
@@ -4756,8 +4751,28 @@ ts_chunk_merge_on_dimension(Chunk *chunk, const Chunk *merge_chunk, int32 dimens
 
 	ts_chunk_constraint_update_slice_id(chunk->fd.id, slice->fd.id, new_slice->fd.id);
 	ChunkConstraints *ccs = ts_chunk_constraints_alloc(1, CurrentMemoryContext);
-	num_ccs =
-		ts_chunk_constraint_scan_by_dimension_slice_id(new_slice->fd.id, ccs, CurrentMemoryContext);
+	ScanIterator iterator =
+		ts_scan_iterator_create(CHUNK_CONSTRAINT, AccessShareLock, CurrentMemoryContext);
+
+	ts_chunk_constraint_scan_iterator_set_slice_id(&iterator, new_slice->fd.id);
+
+	ts_scanner_foreach(&iterator)
+	{
+		bool isnull;
+		Datum d;
+
+		if (slot_attisnull(ts_scan_iterator_slot(&iterator),
+						   Anum_chunk_constraint_dimension_slice_id))
+			continue;
```

Comment:
Is this possible if we're scanning by dimension slice id?

Reply:
No, I guess it is not so I will remove this check. I believe I based this on code from a more general scanning function.